lib/improver/tests/generate_ancillaries/test_GenerateOrographyBandAncils.py

```diff
@@ -219,25 +211,28 @@ def test_high_landband_data(self):
         """test that a correct mask is produced when the land band is
         higher than any land in the test cube."""
         result = GenOrogMasks().gen_orography_masks(
-            self.orography, self.landmask, self.land_key,
+            self.orography, self.landmask,
             self.high_land_threshold)
         self.assertArrayAlmostEqual(result.data, self.exp_high_landmask)
 
-    def test_nothreshold(self):
-        """test the correct exception is raised for key without threshold"""
-        key = 'land'
-        threshold = []
-        with self.assertRaises(ValueError):
-            GenOrogMasks().gen_orography_masks(
-                self.orography, self.landmask, key, threshold)
-
     def test_all_land_points(self):
         """Test that a correct mask is produced when the landsea mask only has
            land points in it."""
         land_mask_cube = self.landmask.copy()
         land_mask_cube.data = np.ones((3, 3))
         result = GenOrogMasks().gen_orography_masks(
-            self.orography, land_mask_cube, self.land_key,
+            self.orography, land_mask_cube,
+            self.valley_threshold)
+        expected_data = np.array([[[1.0, 1.0, 1.0],
+                                   [0.0, 0.0, 0.0],
+                                   [0.0, 0.0, 0.0]]])
+        self.assertArrayAlmostEqual(result.data, expected_data)
+
+    def test_any_surface_type_mask(self):
```

Comment:
Could add an assertion to this test: self.assertEqual(
result.attributes['topographic_zones_include_seapoints'], "True").  I think this would cover the bug we discussed.

Reply:
Added.